drivers/gpio/gpio_max14906.h

```diff
@@ -0,0 +1,321 @@
+	uint8_t reg_raw;
+	struct {
+		uint8_t VINT_UV:1; /* BIT0 */
+		uint8_t V5_UVLO:1;
+		uint8_t VDD_LOW:1;
+		uint8_t VDD_WARN:1;
+		uint8_t VDD_UVLO:1;
+		uint8_t THRMSHUTD:1;
+		uint8_t LOSSGND:1;
+		uint8_t WDOG_ERR:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_opn_wr_en {
+	uint8_t reg_raw;
+	struct {
+		uint8_t OW_OFF_EN1:1; /* BIT0 */
+		uint8_t OW_OFF_EN2:1;
+		uint8_t OW_OFF_EN3:1;
+		uint8_t OW_OFF_EN4:1;
+		uint8_t GDRV_EN1:1;
+		uint8_t GDRV_EN2:1;
+		uint8_t GDRV_EN3:1;
+		uint8_t GDRV_EN4:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_sht_vdd_en {
+	uint8_t reg_raw;
+	struct {
+		uint8_t SH_VDD_EN1:1; /* BIT0 */
+		uint8_t SH_VDD_EN2:1;
+		uint8_t SH_VDD_EN3:1;
+		uint8_t SH_VDD_EN4:1;
+		uint8_t VDD_OV_EN1:1;
+		uint8_t VDD_OV_EN2:1;
+		uint8_t VDD_OV_EN3:1;
+		uint8_t VDD_OV_EN4:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_config_di {
+	uint8_t reg_raw;
+	struct {
+		uint8_t OVL_BLANK:2; /* BIT0 */
+		uint8_t OVL_STRETCH_EN:1;
+		uint8_t ABOVE_VDD_PROT_EN:1;
+		uint8_t VDD_FAULT_SEL:1;
+		uint8_t VDD_FAULT_DIS:1;
+		uint8_t RESERVED:1;
+		uint8_t TYP_2_DI:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_config_do {
+	uint8_t reg_raw;
+	struct {
+		uint8_t DO_MODE1:2; /* BIT0 */
+		uint8_t DO_MODE2:2;
+		uint8_t DO_MODE3:2;
+		uint8_t DO_MODE4:2; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_config_curr_lim {
+	uint8_t reg_raw;
+	struct {
+		uint8_t CL1:2; /* BIT0 */
+		uint8_t CL2:2;
+		uint8_t CL3:2;
+		uint8_t CL4:2; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_mask {
+	uint8_t reg_raw;
+	struct {
+		uint8_t OVER_LD_M:1; /* BIT0 */
+		uint8_t CURR_LIM_M:1;
+		uint8_t OW_OFF_M:1;
+		uint8_t ABOVE_VDD_M:1;
+		uint8_t SHT_VDD_M:1;
+		uint8_t VDD_OK_M:1;
+		uint8_t SUPPLY_ERR_M:1;
+		uint8_t COM_ERR_M:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_config1 {
+	uint8_t reg_raw;
+	struct {
+		uint8_t FLED_SET:1; /* BIT0 */
+		uint8_t SLED_SET:1;
+		uint8_t FLED_STRETCH:2;
+		uint8_t FFILTER_EN:1;
+		uint8_t FILTER_LONG:1;
+		uint8_t FLATCH_EN:1;
+		uint8_t LED_CURR_LIM:1; /* BIT7 */
+	} reg_bits;
+};
+
+union max14906_config2 {
+	uint8_t reg_raw;
+	struct {
+		uint8_t VDD_ON_THR:1; /* BIT0 */
+		uint8_t SYNCH_WD_EN:1;
+		uint8_t SHT_VDD_THR:2;
+		uint8_t OW_OFF_CS:2;
+		uint8_t WD_TO:2; /* BIT7 */
+	} reg_bits;
+};
+
+/* Config1 register Enable/Disable SLED */
+#define MAX149x6_SLED_MASK	BIT(1)
+/* Config1 register Enable/Disable FLED */
+#define MAX149x6_FLED_MASK	BIT(0)
+
+#define MAX149x6_ENABLE 1
+#define MAX149x6_DISABLE 0
+
+struct max149x6_config {
+	struct spi_dt_spec spi;
+	struct gpio_dt_spec fault_gpio;
+	struct gpio_dt_spec ready_gpio;
+	struct gpio_dt_spec sync_gpio;
+	struct gpio_dt_spec en_gpio;
+	bool crc_en;
+	union max14906_config1 config1;
+	union max14906_config2 config2;
+	union max14906_config_curr_lim curr_lim;
+	union max14906_config_di config_do;
+	union max14906_config_do config_di;
+	enum max149x6_spi_addr spi_addr;
+	uint8_t pkt_size;
+};
+
+#define max14906_config max149x6_config
```

Comment:
Is there a specific reason not using typedef?

Reply:
Mainly because of @MaureenHelm comment here https://github.com/zephyrproject-rtos/zephyr/pull/71141#discussion_r1702384039
I personally prefer them but since they are not very welcomed in Zephyr that is fine by me.

Reply:
Thanks to point it.